numpy/lib/function_base.py

```diff
@@ -2669,6 +2689,19 @@ class vectorize(object):
            <http://docs.scipy.org/doc/numpy/reference/c-api.generalized-ufuncs.html>`_.
     """
 
+    # __new__ is overriden as a clean way to provide decorator-with-keywords
+    # syntax while allowing 'vectorize' to remain as the type of vectorized
+    # functions.
+    def __new__(cls, pyfunc=np._NoValue, *args, **kwargs):
```

Comment:
oops, that's a documentation bug (because I originally wrote this as `None`), but I also agree with your reasoning.  Waiting to hear if eric had a compelling reason for suggesting `_NoValue` here